src/utils/workflow_utils/src/workflow_utils.c

```diff
@@ -1804,6 +1816,107 @@ bool workflow_set_sandbox(ADUC_WorkflowHandle handle, const char* sandbox)
     return true;
 }
 
+bool _workflow_cpy_config_download_folder(char* dest, const size_t dest_size)
+{
+    bool success = false;
+    const ADUC_ConfigInfo* config = ADUC_ConfigInfo_GetInstance();
+    if (config == NULL)
+    {
+        goto done; // could not get
+    }
+
+    size_t download_folder_len = ADUC_StrNLen(config->downloadsFolder, dest_size);
```

Comment:
you want strlen, not strnlen.  if you pass strnlen then you won't get larger than dest_size which means that you'll never get an error from safestrcopyn on 1836 which means you'll truncate the path which is **not** desirable.